test/evp_extra_test.c

```diff
@@ -2891,6 +2980,388 @@ static int test_names_do_all(void)
+        errmsg = "WRONG_RESULT (reinit)";
+        goto err;
+    }
+    testresult = 1;
+ err:
+    if (errmsg != NULL)
+        TEST_info("test_evp_reset %d: %s", idx, errmsg);
+    EVP_CIPHER_CTX_free(ctx);
+    EVP_CIPHER_free(type);
+    return testresult;    
+}
+
+typedef struct {
+    const unsigned char *iv1;
+    const unsigned char *iv2;
+    const unsigned char *expected1;
+    const unsigned char *expected2;
+    const unsigned char *tag1;
+    const unsigned char *tag2;
+    size_t ivlen1;
+    size_t ivlen2;
+    size_t expectedlen1;
+    size_t expectedlen2;
+} TEST_GCM_IV_REINIT_st;
+
+static const TEST_GCM_IV_REINIT_st gcm_reinit_tests[] = {
+    {
+        iGCMResetIV1, iGCMResetIV2, gcmResetCiphertext1, gcmResetCiphertext2,
+        gcmResetTag1, gcmResetTag2, sizeof(iGCMResetIV1), sizeof(iGCMResetIV2),
+        sizeof(gcmResetCiphertext1), sizeof(gcmResetCiphertext2)
+    },
+    {
+        iGCMResetIV2, iGCMResetIV1, gcmResetCiphertext2, gcmResetCiphertext1,
+        gcmResetTag2, gcmResetTag1, sizeof(iGCMResetIV2), sizeof(iGCMResetIV1),
+        sizeof(gcmResetCiphertext2), sizeof(gcmResetCiphertext1)
+    }
+};
+
+static int test_gcm_reinit(int idx)
+{
+    int outlen1, outlen2, outlen3;
+    int testresult = 0;
+    unsigned char outbuf[1024];
+    unsigned char tag[16];
+    const TEST_GCM_IV_REINIT_st *t = &gcm_reinit_tests[idx];
+    EVP_CIPHER_CTX *ctx = NULL;
+    EVP_CIPHER *type = NULL;
+    char *errmsg = NULL;
+
+    ctx = EVP_CIPHER_CTX_new();
+    if (ctx == NULL) {
+        errmsg = "CTX_ALLOC";
+        goto err;
+    }
+    type = EVP_CIPHER_fetch(testctx, "aes-256-gcm", testpropq);
+    if (type == NULL) {
+        errmsg = "CIPHER_FETCH";
+        goto err;
+    }
+    if (!EVP_CipherInit_ex(ctx, type, NULL, NULL, NULL, 1)) {
+        errmsg = "ENC_INIT";
+        goto err;
+    }
+    if (!EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_GCM_SET_IVLEN, t->ivlen1, NULL)) {
+        errmsg = "SET_IVLEN1";
+        goto err;
+    }
+    if (!EVP_CipherInit_ex(ctx, NULL, NULL, kGCMResetKey, t->iv1, 1)) {
+        errmsg = "SET_IV1";
+        goto err;
+    }
+    if (!EVP_CipherUpdate(ctx, NULL, &outlen3, gcmAAD, sizeof(gcmAAD))) {
+        errmsg = "AAD1";
+        goto err;
+    }
+    EVP_CIPHER_CTX_set_padding(ctx, 0);
+    if (!EVP_CipherUpdate(ctx, outbuf, &outlen1, gcmResetPlaintext,
+                          sizeof(gcmResetPlaintext))) {
+        errmsg = "CIPHER_UPDATE1";
+        goto err;
+    }
+    if (!EVP_CipherFinal_ex(ctx, outbuf + outlen1, &outlen2)) {
+        errmsg = "CIPHER_FINAL1";
+        goto err;
+    }
+    if (!TEST_mem_eq(t->expected1, t->expectedlen1, outbuf, outlen1 + outlen2)) {
+        errmsg = "WRONG_RESULT1";
+        goto err;
+    }
+    if (!EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_GET_TAG, 16, tag)) {
+        errmsg = "GET_TAG1";
+        goto err;
+    }
+    if (!TEST_mem_eq(t->tag1, 16, tag, 16)) {
+        errmsg = "TAG_ERROR1";
+        goto err;
+    }
+    // Now reinit
```

Comment:
No C++ style comments allowed: Use /* */

Reply:
This is the cause of the test failure.

Reply:
Didn't know that.  Sorry.  Fixed.